pmd-core/src/main/java/net/sourceforge/pmd/renderers/Renderer.java

```diff
@@ -163,4 +164,17 @@
     void end() throws IOException;
 
     void flush() throws IOException;
+
+    /**
+     * Sets the filename where the report should be written to. If no filename is provided,
+     * the renderer should write to stdout.
+     *
+     * <p>Implementations must initialize the writer of the renderer.
+     *
+     * <p>See {@link AbstractRenderer#setReportFile(String)} for the default impl.
+     *
+     * @param reportFilename the filename (optional).
+     */
+    @Experimental
+    void setReportFile(String reportFilename);
```

Comment:
In a mature API, I think this should take a `Path` argument instead of a string. But then, everything should be updated to be consistent